src/sql-parser/src/parser.rs

```diff
@@ -1915,6 +1927,67 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_kafka_security(&mut self) -> Result<KafkaSecurityOptions, ParserError> {
+        Ok(match self.expect_one_of_keywords(&[SSL, SASL, SASLSSL])? {
+            Keyword::Ssl => {
+                let (mut key, mut certificate, mut passphrase) = (None, None, None);
+                while let Ok(keyword) = self.expect_one_of_keywords(&[KEY, CERTIFICATE, PASSWORD]) {
+                    match keyword {
+                        Keyword::Key => {
+                            let _ = self.consume_token(&Token::Eq);
+                            key = Some(self.parse_object_name()?);
+                        }
+                        Keyword::Certificate => {
+                            let _ = self.consume_token(&Token::Eq);
+                            certificate = Some(self.parse_object_name()?);
+                        }
+                        Keyword::Password => {
+                            let _ = self.consume_token(&Token::Eq);
+                            passphrase = Some(self.parse_object_name()?);
+                        }
+                        _ => unreachable!(),
+                    }
+                }
+                KafkaSecurityOptions::SSL {
+                    key,
+                    certificate,
+                    passphrase,
+                }
+            }
+            Keyword::Sasl => self.parse_kafka_sasl(false)?,
+            Keyword::SaslSsl => self.parse_kafka_sasl(true)?,
+            _ => unreachable!(),
+        })
+    }
+
+    fn parse_kafka_sasl(&mut self, ssl: bool) -> Result<KafkaSecurityOptions, ParserError> {
+        self.expect_keyword(MECHANISM)?;
```

Comment:
This is likely just my own ignorance--where does the convention come from to prescribe the order the options must be input?

Reply:
Typically for SASL definitions `MECHANISM` has to come first since not every mechanism supports the same options. In our case all three we support are username/password based, so we could get away with allowing them in any order I guess.

Reply:
I would prefer to make that flexible in the parser rather than forcing a strict order.